### ISSUE TITLE
Item List does not immediately scrollto selection.

### ISSUE THREAD

**Issue by Brickcaster** [godotengine/godot#6496]
**Operating system or device - Godot version:**
Windows  - 2.1

**Issue description** (what happened, and what was expected):
I created an ItemList with a large number of elements.  I select one and press up-down to scroll out of the display bounds.  The scrollbar moves but the selection remains outside of the box.  After waiting one second, the list scrolls back to bring the selection into visibility.

** Steps to reproduce **
Run attached project.
Select item in list.
Press up/down arrows until selection is out of display bounds of ItemList.
Observe scroll bar moves immediately but display does not.
Observe that after a brief delay, the box display moves to bring the selected item in focus.

**Link to minimal example project** (optional but very welcome):
https://www.dropbox.com/s/mz1gb2k0oyq0cxf/Godot%202.1%20ItemListScrolling%20bug%20demo.zip?dl=0

**Comment by Brickcaster** [comment]
I can't seem to reproduce the "after brief delay" step.  However another issue I noticed is:

Reversing direction scrolls to bring the previously selected item in focus.
That is to say, in the example project:

12 is the last visible item in ItemList.
12 is selected.
Press down arrow.
13 is selected.
Scrollbar was updated and moves down.
13 is not visible. **Unexpected behavior**
Visible area did not change. **Unexpected behavior**
Press up arrow.
12 is selected.
13 is now visible. **Unexpected behavior**
Scrollbar was not updated.

In my testing, changing when ensure_current_is_visible() is called does not seem to fix the issue.  I will investigate ensure_current_is_visible() itself.
